src/detect-pcre.c

```diff
@@ -124,33 +138,27 @@ void DetectPcreRegister (void)
         }
     }
 
-    DetectSetupParseRegexes(PARSE_REGEX, &parse_regex, &parse_regex_study);
-
-    /* setup the capture regex, as it needs PCRE_UNGREEDY we do it manually */
-    const char *eb;
-    int eo;
-    int opts = PCRE_UNGREEDY; /* pkt_http_ua should be pkt, http_ua, for this reason the UNGREEDY */
-
-    parse_capture_regex = pcre_compile(PARSE_CAPTURE_REGEX, opts, &eb, &eo, NULL);
-    if (parse_capture_regex == NULL)
-    {
-        FatalError(SC_ERR_PCRE_COMPILE, "pcre compile of \"%s\" failed at offset %" PRId32 ": %s", PARSE_CAPTURE_REGEX, eo, eb);
-    }
-
-    parse_capture_regex_study = pcre_study(parse_capture_regex, 0, &eb);
-    if(eb != NULL)
-    {
-        FatalError(SC_ERR_PCRE_STUDY, "pcre study failed: %s", eb);
-    }
-
 #ifdef PCRE_HAVE_JIT
     if (PageSupportsRWX() == 0) {
         SCLogConfig("PCRE won't use JIT as OS doesn't allow RWX pages");
         pcre_use_jit = 0;
     }
+
+    DetectSetupParseRegexesOptsWithJIT(PARSE_REGEX, &parse_regex, 0, pcre_use_jit == 1);
+#else
+    DetectSetupParseRegexes(PARSE_REGEX, &parse_regex);
 #endif
 
-    DetectParseRegexAddToFreeList(parse_capture_regex, parse_capture_regex_study);
+    /* setup the capture regex, as it needs PCRE_UNGREEDY we do it manually */
+    int opts = PCRE_UNGREEDY; /* pkt_http_ua should be pkt, http_ua, for this reason the UNGREEDY */
+
+#ifdef PCRE_HAVE_JIT
+    DetectSetupParseRegexesOptsWithJIT(PARSE_CAPTURE_REGEX, &parse_capture_regex, opts, pcre_use_jit == 1);
```

Comment:
whats the purpose of this call?

Reply:
This creates the structure for a JIT capable exec call. Should that be used for the capture regex?

Reply:
My understanding, which may be outdated, is that the jit support and capturing substrings are mutually exclusive. Jit just doesn't support it. If this is still correct, and it should be easy to test, then having jit support in the pcre for rule parsing doesn't seem to make much sense.

It does make a lot of sense for runtime packet matching, as there we don't often do substring capture.

Reply:
JIT *can be* used for capturing substrings but the documentation states 

```
Just-in-time compiling is a heavyweight optimization that can greatly speed up pattern matching. However, it comes at the cost of extra processing before the match is performed. Therefore, it is of most benefit when the same pattern is going to be matched many times.
```

Thus, I'll change this in #4377 to use `DetectSetupParseRegexes` (no jit)